builtin/logical/transit/backend.go

```diff
@@ -100,13 +113,45 @@ func GetCacheSizeFromStorage(ctx context.Context, s logical.Storage) (int, error
 	return size, nil
 }
 
-func (b *backend) invalidate(_ context.Context, key string) {
+// Update cache size and get policy
+func (b *backend) GetPolicy(ctx context.Context, polReq keysutil.PolicyRequest, rand io.Reader) (retP *keysutil.Policy, retUpserted bool, retErr error) {
+	if b.lm.GetUseCache() && b.cacheSizeChanged {
+		var err error
+		currentCacheSize := b.lm.GetCacheSize()
+		storedCacheSize, err := GetCacheSizeFromStorage(ctx, polReq.Storage)
+		if err != nil {
+			return nil, false, err
+		}
+		if currentCacheSize != storedCacheSize {
+			err = b.lm.RefreshCache(storedCacheSize)
+			if err != nil {
+				return nil, false, err
+			}
+		}
+		// Acquire the lock to modify cacheSizeChanged
+		b.configMutex.Lock()
+		defer b.configMutex.Unlock()
+		b.cacheSizeChanged = false
```

Comment:
There is a bug here where if the invalidation happens, but someone calls GetPolicy before the storage change actually arrives over replication, the cache sizes won't have changed but you'll have cleared cacheSizeChanged.  Only reset this if you changed the size.